src/main/java/domain/Invoice.kt

```diff
@@ -15,20 +15,25 @@ import javax.persistence.Id
 import javax.persistence.ManyToOne
 import javax.persistence.NamedQuery
 import javax.persistence.Table
+import javax.persistence.Temporal
+import javax.persistence.TemporalType
 
 @Entity
 @NamedQuery(name = "Invoice.allInvoices", query = "SELECT i FROM Invoice i")
 @Table(name = "invoice")
 data class Invoice(
 
     @Enumerated(EnumType.STRING)
-    var generationType: InvoiceGenerationType,
+    val generationType: InvoiceGenerationType,
     @Enumerated(EnumType.STRING)
     var state: InvoiceState = InvoiceState.OPEN,
     @ManyToOne
-    var owner: Profile? = null,
+    val profile: Profile? = null,
     @ManyToOne
-    var vehicle: Vehicle? = null
+    var vehicle: Vehicle? = null,
+    @Temporal(TemporalType.DATE)
+    val expires: Date = now(),
```

Comment:
Why is this in the contructor when it's value is defaulted?

Reply:
Fixed